### PR TITLE
Add remove poster function

### PR DESCRIPTION
**Pull request by Ryan-D-Miller**
## Is this a fix or a feature?
* Feature
## What is the change?
* Being able to delete posters from saved poster
## What does it fix?
* fixed saved poster function
## Where should the reviewer start?
* Ln 46
## How should this be tested? 
* In the browser

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -101,3 +107,10 @@ function checkDuplicate () {
   }
   return false;
 }
+
+function removePoster(target){
+  console.log(savedPosters);
```
Comment (daomeow) [inline src/main.js]: ** **Note to self** ** remove console log

File: src/main.js [hunk 2]
```diff
@@ -43,6 +43,12 @@ takeMeBackButton.addEventListener('click', function(){
 
 savePosterButton.addEventListener('click', savePoster);
 
+document.addEventListener('click', function(e){
+  if(e.target && e.target.parentElement.classList.contains("saved-poster")){
+    removePoster(e.target);
+  }
```
Comment (HartiganHM) [inline src/main.js]: This suggestion will have to be made with my next comment. If you rewrite this to pass `e.target.parentElement` (which would be the `article` element), you can...

File: src/main.js [hunk 3]
```diff
@@ -101,3 +107,10 @@ function checkDuplicate () {
   }
   return false;
 }
+
+function removePoster(target){
+  console.log(savedPosters);
+  savedPosters.splice(target.parentElement.id, 1);
+  target.parentElement.parentElement.removeChild(target.parentElement);
```
Comment (HartiganHM) [inline src/main.js]: Refactor these two lines assuming that the `target` parameter is the `article` element (which you want to remove) and can refactor this to the below:

```javascript
savedPosters.splice(target.id, 1);
target.remove();
```

File: src/main.js (hunk 1 above)
Comment (HartiganHM) [reply src/main.js]: Yes, definitely make sure these get pulled out _before_ the PR is merged 🙂

File: src/main.js [hunk 4]
```diff
@@ -84,7 +90,7 @@ function savePoster() {
   var duplicate = checkDuplicate();
   if (!duplicate) {
       savedPosters.push(currentPoster);
-      var newPoster = `${savedPosterLocation.innerHTML} <article class="poster">
+      var newPoster = `${savedPosterLocation.innerHTML} <article class="poster saved-poster" id=${savedPosters.length -1}>
```
Comment (HartiganHM) [inline src/main.js]: Standard formatting will have a space between the minus sign and the one: `savedPosters.length - 1`.

**Review by HartiganHM — COMMENTED** [review]
This is looking pretty good! I think there's an opportunity to refactor the way you're removing the poster. There's a lot of bouncing around with "parent" and "children".

Also, the styles when viewing "Saved Posters" is a little off. I would try to fix this before you turn in the project 🙂 

image

**Comment by daomeow** [comment]
@HartiganHM thank you for your feedback! Looking into these changes now :)

File: src/main.js (hunk 2 above)
Comment (daomeow) [reply src/main.js]: @Ryan-D-Miller in our `.poster-img` class (style.css line 51) - please remove `height: 450px` and `width: 800px`. Add in `max-width: 80%;`

**Pull request merged**